oneDAL development env docker image (#2513)
oneapi-src/oneDAL: INSTALL.md

For details, see [System Requirements for oneDAL](https://www.intel.com/content/www/us/en/developer/articles/system-requirements/system-requirements-for-oneapi-data-analytics-library.html).

> ## Docker development environement

Reviewer: ```suggestion
## Docker Development Environment 
```

> [Docker file](https://github.com/oneapi-src/oneDAL/tree/master/dev) with oneDAL development environemnt

Reviewer: ```suggestion
[Docker file](https://github.com/oneapi-src/oneDAL/tree/master/dev) with the oneDAL development environment
```

> [Docker file](https://github.com/oneapi-src/oneDAL/tree/master/dev) with the oneDAL development environment

Reviewer 2: ```suggestion
[Docker file](https://github.com/oneapi-src/oneDAL/tree/master/dev/docker) with the oneDAL development environment
```

> is avaialbe as alternative for setting up environemnt manually. 

Reviewer: ```suggestion
is available as an alternative to the manual setup. 
```
